www/localPackage.ts

```diff
@@ -71,41 +71,85 @@ class LocalPackage extends Package implements ILocalPackage {
                 this.finishInstall(deployDir, installOptions, installSuccess, installError);
             };
 
-            var newPackageUnzipped = (unzipError: any) => {
-                if (unzipError) {
-                    installError && installError(new Error("Could not unzip package. " + CodePushUtil.getErrorMessage(unzipError)));
+            var newPackageUnzippedAndVerified: Callback<boolean> = (error) => {
+                if (error) {
+                    installError && installError(new Error("Could not unzip and verify package. " + CodePushUtil.getErrorMessage(error)));
                 } else {
                     LocalPackage.handleDeployment(newPackageLocation, CodePushUtil.getNodeStyleCallbackFor<DirectoryEntry>(donePackageFileCopy, installError));
                 }
             };
 
             FileUtil.getDataDirectory(LocalPackage.DownloadUnzipDir, false, (error: Error, directoryEntry: DirectoryEntry) => {
-                var unzipPackage = () => {
+                var unzipAndVerifyPackage = () => {
                     FileUtil.getDataDirectory(LocalPackage.DownloadUnzipDir, true, (innerError: Error, unzipDir: DirectoryEntry) => {
                         if (innerError) {
                             installError && installError(innerError);
-                        } else {
-                            zip.unzip(this.localPath, unzipDir.toInternalURL(), newPackageUnzipped);
+                            return;
                         }
+
+                        zip.unzip(this.localPath, unzipDir.toInternalURL(), (unzipError: any) => {
+                            if (unzipError) {
+                                installError && installError(new Error("Could not unzip package. " + CodePushUtil.getErrorMessage(unzipError)));
+                            }
+                           this.verifyPackage(unzipDir, installError, newPackageUnzippedAndVerified);
+                        });
+
                     });
                 };
 
                 if (!error && !!directoryEntry) {
                     /* Unzip directory not clean */
                     directoryEntry.removeRecursively(() => {
-                        unzipPackage();
+                        unzipAndVerifyPackage();
                     }, (cleanupError: FileError) => {
                         installError && installError(FileUtil.fileErrorToError(cleanupError));
                     });
                 } else {
-                    unzipPackage();
+                    unzipAndVerifyPackage();
                 }
             });
         } catch (e) {
             installError && installError(new Error("An error occured while installing the package. " + CodePushUtil.getErrorMessage(e)));
         }
     }
 
+    private verifyPackage(unzipDir: DirectoryEntry, installError: ErrorCallback, callback: Callback<boolean>): void {
+        var packageHashSuccess = (localHash: string) => {
```

Comment:
We should also log local hash and package hash like this: log("Expected hash: " + this.packageHash + ", actual hash: " + localHash);